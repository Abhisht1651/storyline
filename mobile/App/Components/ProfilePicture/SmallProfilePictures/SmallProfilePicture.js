// @flow

import React from 'react'
import { TouchableOpacity, Image, View } from 'react-native'
import Images from '../../../Themes/Images'
import styles from '../../Styles/ProfilePictureStyles/SmallProfilePictureStyles/SmallProfilePictureStyle'

type Props = {
  image: Image,
  type: number,
  onPress: () => {}
}

const SmallProfilePicture = (props: Props) => {
  const { image, type, onPress } = props
  let component = {}
  if (type === 1) {
    component = (
      <TouchableOpacity style={styles.container1} activeOpacity={1} onPress={onPress}>
        <Image source={image} style={styles.image} />
      </TouchableOpacity>
    )
  } else {
    component = (
      <TouchableOpacity style={styles.container2} activeOpacity={1} onPress={onPress}>
        <Image source={image} style={styles.image} />
      </TouchableOpacity>
    )
  }
  return <View>{component}</View>
}
SmallProfilePicture.defaultProps = {
  image: Images.profilePicture,
  type: 1,
  onPress: () => {}
}

export default SmallProfilePicture
