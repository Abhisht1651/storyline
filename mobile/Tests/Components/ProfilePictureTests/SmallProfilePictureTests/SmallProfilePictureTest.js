import { shallow } from 'enzyme'
import React from 'react'
import SmallProfilePicture from '../../../../App/Components/ProfilePicture/SmallProfilePictures/SmallProfilePicture'

describe('SmallProfilePicture1', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<SmallProfilePicture />)
      expect(component).toMatchSnapshot()
    })
  })
})
