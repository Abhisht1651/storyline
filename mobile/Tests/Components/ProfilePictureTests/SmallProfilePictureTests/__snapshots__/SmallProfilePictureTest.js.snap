// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallProfilePicture1 Rendering should match to snapshot 1`] = `
<View>
  <ForwardRef
    activeOpacity={1}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "borderRadius": 100,
        "justifyContent": "center",
        "width": 45.622142857142855,
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "aspectRatio": 1,
          "borderRadius": 100,
          "height": "100%",
        }
      }
    />
  </ForwardRef>
</View>
`;
